ldes-fragmentisers/ldes-fragmentisers-pagination/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/pagination/MemberPaginationService.java

```diff
@@ -0,0 +1,94 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.fragmentisers.pagination;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.events.fragmentation.MemberAllocatedEvent;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.LdesFragmentIdentifier;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.ViewName;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.entities.BucketisedMember;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.entities.Fragment;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.entities.FragmentSequence;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.repository.BucketisedMemberRepository;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.repository.FragmentRepository;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.repository.FragmentSequenceRepository;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentisers.pagination.services.OpenPageProvider;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.springframework.context.ApplicationEventPublisher;
+
+import java.util.List;
+import java.util.concurrent.Future;
+
+public class MemberPaginationService {
+    private final FragmentSequenceRepository fragmentSequenceRepository;
+    private final BucketisedMemberRepository bucketisedMemberRepository;
+    private final OpenPageProvider openPageProvider;
+    private final FragmentRepository fragmentRepository;
+    private final ApplicationEventPublisher eventPublisher;
+    private final ViewName viewName;
+    private long sequenceNr;
+    private Future<?> task;
+
+    public MemberPaginationService(FragmentSequenceRepository fragmentSequenceRepository, BucketisedMemberRepository bucketisedMemberRepository,
+                                   OpenPageProvider openPageProvider, FragmentRepository fragmentRepository, ApplicationEventPublisher eventPublisher, ViewName viewName) {
+        this.fragmentSequenceRepository = fragmentSequenceRepository;
+        this.bucketisedMemberRepository = bucketisedMemberRepository;
+        this.openPageProvider = openPageProvider;
+        this.fragmentRepository = fragmentRepository;
+        this.eventPublisher = eventPublisher;
+        this.viewName = viewName;
+        sequenceNr = determineLastProcessedSequence(viewName).sequenceNr();
+    }
+
+    public void paginateMember() {
+        List<BucketisedMember> bucketisedMembers = getNextMember(viewName);
+        while(!bucketisedMembers.isEmpty()) {
+            List<ImmutablePair<Fragment, BucketisedMember>> pages = getPages(bucketisedMembers);
+            allocateMembers(pages);
+
+            fragmentSequenceRepository.saveLastProcessedSequence(new FragmentSequence(viewName, sequenceNr + 1));
+            sequenceNr += 1;
```

Comment:
if you do sequenceNr++, the increment will not be added before the computation of the FragmentSequence i believe

Reply:
I believe so too, however, instead of `sequenceNr++`, `++sequenceNr` could be used, which should do the addition beforehand